src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SnsService.kt

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * SQS Reference
+ * https://github.com/ministryofjustice/hmpps-spring-boot-sqs
+ */
+
+@Service
+class SnsService(hmppsQueueService: HmppsQueueService, private val objectMapper: ObjectMapper) {
+
+  private val domaineventsTopic by lazy { hmppsQueueService.findByTopicId(TOPIC_ID) ?: throw RuntimeException("Topic with name $TOPIC_ID doesn't exist") }
+  private val domaineventsTopicClient by lazy { domaineventsTopic.snsClient }
+
+  fun sendVisitBookedEvent(visit: VisitDto) {
+    log.debug("Sending visit booked event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_BOOKED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_BOOKED_DESC,
+        occurredAt = Instant.now().toString(),
+        prisonerId = visit.prisonerId,
+        additionalInformation = AdditionalInformation(
+          reference = visit.reference,
+        )
+      )
+    )
+  }
+
+  fun sendVisitCancelledEvent(visit: VisitDto) {
+    log.debug("Sending visit cancelled event")
+    publishToDomainEventsTopic(
+      HMPPSDomainEvent(
+        eventType = EVENT_PRISON_VISIT_CANCELLED,
+        version = EVENT_PRISON_VISIT_VERSION,
+        description = EVENT_PRISON_VISIT_CANCELLED_DESC,
+        occurredAt = Instant.now().toString(),
+        prisonerId = visit.prisonerId,
+        additionalInformation = AdditionalInformation(
+          reference = visit.reference
+        )
+      )
+    )
+  }
+
+  private fun publishToDomainEventsTopic(payload: HMPPSDomainEvent) {
+    log.debug("Event ${payload.eventType} for id ${payload.additionalInformation.reference}")
+
+    try {
+      domaineventsTopicClient.publish(
+        PublishRequest(domaineventsTopic.arn, objectMapper.writeValueAsString(payload))
+          .withMessageAttributes(
+            mapOf(
+              "eventType" to MessageAttributeValue().withDataType("String").withStringValue(payload.eventType)
+            )
+          ).also { log.info("Published event $payload to outbound topic") }
+      )
+    } catch (e: Throwable) {
+      // throw PublishEventException("Failed to publish Event $payload.eventType to $TOPIC_ID", e)
+
+      // Note: Silently fail until VB-671 is implemented
+      log.debug("Failed to publish Event $payload.eventType to $TOPIC_ID", e)
+    }
+  }
+
+  companion object {
+    const val TOPIC_ID = "domainevents"
+    const val EVENT_ZONE_ID = "Europe/London"
+    const val EVENT_PRISON_VISIT_VERSION = 1
+    const val EVENT_PRISON_VISIT_BOOKED = "prison-visit.booked"
+    const val EVENT_PRISON_VISIT_BOOKED_DESC = "Prison Visit Booked"
+    const val EVENT_PRISON_VISIT_CANCELLED = "prison-visit.cancelled"
+    const val EVENT_PRISON_VISIT_CANCELLED_DESC = "Prison Visit Cancelled"
+
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+}
+
+data class AdditionalInformation(
+  val reference: String,
+)
+
+@Suppress("unused")
+data class HMPPSDomainEvent(
```

Comment:
DTO, do you like internal classes?

Reply:
Ill review the encapsulation, like you say its a serialized object we might want to expose it.